prov/util/src/util_wait.c

```diff
@@ -229,7 +229,7 @@ static int util_wait_fd_try(struct util_wait *wait)
 	dlist_foreach_container(&wait_fd->fd_list, struct ofi_wait_fd_entry,
 				fd_entry, entry) {
 		ret = fd_entry->try(fd_entry->arg);
-		if (ret != FI_SUCCESS) {
+		if (ret != FI_SUCCESS && ret != -FI_EAGAIN) {
```

Comment:
Trywait doesn't guarantee progress.  It only needs to indicate if it's safe to block or not.  Assuming that the tr() callbacks follow the same semantic as trywait, getting EAGAIN here should be enough to know that it's not safe to block.